### ISSUE TITLE
str_concat produces single element Lists in groupby context

### ISSUE THREAD

**Issue by KevDAll** [pola-rs/polars#2481]
#### Are you using Python or Rust?

Python

#### Which feature gates did you use?

NA

#### What version of polars are you using?

0.12.8 (verified in 0.12.17)

#### What operating system are you using polars on?

Debian (python:3.8 base docker, verified in windows)

#### Describe your bug.

When used in the select context, expr.str_concat works as expected and requested in #1490, and produces a UTF8 value. However, in an aggregation context, the value is produced correctly, but instead of being a UTF8 column, the result is a column of arrays with a length of 1. This requires a python eval expression  and copy to convert into a UTF8 column.

#### What are the steps to reproduce the behavior?
``` python
import unittest

import polars as pl


class PolarsStrConcatTest(unittest.TestCase):
    #This test passes
    def test_str_concat_select(self):
        """str_concat in select context works as expected"""
        test_pl = pl.DataFrame([pl.Series('val', ['A', 'B', 'C', 'D'])])
        expr = pl.col('val').str_concat(delimiter=',')
        grouped = test_pl.select(expr.alias('grouped')).get_column('grouped')
        self.assertEqual(grouped.dtype, pl.Utf8)
    
    #This test fails
    def test_str_concat_groupby(self):
        """str_concat in aggregation context produces arrays of length 1"""
        test_pl = pl.DataFrame([pl.Series('val', ['A','B','C','D']),
                                pl.Series('id', [1, 1, 2, 2])])
        expr = pl.col('val').str_concat(delimiter=',')
        grouped = test_pl.groupby('id').agg(expr.alias('grouped')).get_column('grouped')
        self.assertEqual(grouped.dtype, pl.Utf8)
 ```

Output:
```
test_str_concat_groupby (polars_group_concat_test.PolarsStrConcatTest)
str_concat in aggregation context produces arrays of length 1 ... FAIL
test_str_concat_select (polars_group_concat_test.PolarsStrConcatTest)
str_concat in select context works as expected ... ok

======================================================================
FAIL: test_str_concat_groupby (polars_group_concat_test.PolarsStrConcatTest)
str_concat in aggregation context produces arrays of length 1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/unit_tests/polars_group_concat_test.py", line 22, in test_str_concat_groupby
    self.assertEqual(grouped.dtype, pl.Utf8)
AssertionError: <class 'polars.datatypes.List'> != <class 'polars.datatypes.Utf8'>

----------------------------------------------------------------------
Ran 2 tests in 0.002s

FAILED (failures=1)
```

#### What is the expected behavior?

The second test case should pass and produce a Utf8 column instead of a column of Lists each with length 1.

**Comment by ritchie46** [comment]
Thanks.. will fix this.


> This requires a python eval expression and copy to convert into a UTF8 column

Try `flatten()` 😉

**Comment by ritchie46** [comment]
Fixed by #2482, note that the exprs `str_concat` is moved to `str.concat`. I don't know why it wasn't there in the first place.
